iroh-gateway/src/main.rs

```diff
@@ -10,12 +10,21 @@ use iroh_gateway::{
     metrics,
 };
 use iroh_rpc_client::Client as RpcClient;
+use iroh_util::lock::ProgramLock;
 use iroh_util::{iroh_config_path, make_config};
 use tokio::sync::RwLock;
 use tracing::{debug, error};
 
 #[tokio::main(flavor = "multi_thread")]
 async fn main() -> Result<()> {
+    let mut lock = ProgramLock::new("iroh-gateway")?;
+    if lock.is_locked() {
+        println!("iroh-gateway is already running, stopping.");
+        return Ok(());
```

Comment:
I think this should exit with a non-zero status. and should print in a nice pretty yellow

Reply:
- `eprintln` to print to stderr
- https://doc.rust-lang.org/std/process/fn.exit.html for exiting